### PR TITLE
Issue 644 - License

### LINKED ISSUE #644: Verify licenses of 3rd party software are compatible with MIT
LUE is licensed under the MIT license. This is only possible if licenses of software we depend on are compatible. This needs to be verified. Also, if 3rd party licenses require us to ship their license and copyright notices, we must start doing this.

### PR DESCRIPTION
**Pull request by saeb-faraji-gargari**
Closes #644

### CODE REVIEW

File: LICENSE.txt
Comment (kordejong) [inline LICENSE.txt:1]: You can leave this file unchanged. It is our license. You don't need to refer to licenses of software we use.

File: document/doc/code/index.rst
Comment (kordejong) [inline document/doc/code/index.rst:1]: - Please remove gperftools. It is a dependency of HPX.
- You may add that the libraries are used, depending on the targets built. (For example, Pybind11 is not used if the Python bindings are not built. GLFW, Vulkan and ImGui are not used if lue_view is not built.)
- Replace sentence starting with "Licenses for third-..." with "The licenses for these libraries can be found in the license directory of the LUE source repository." Maybe turn "license" in this sentence into a link.

**Review by kordejong — CHANGES_REQUESTED** [review]
Thanks Saeb. I have some minor comments. Once handled  I will merge the PR.

File: document/doc/code/index.rst
Comment (saeb-faraji-gargari) [reply document/doc/code/index.rst:1]: Please check the following table to list the libraries based on target builds

LUE_BUILD_DATA_MODEL and LUE_BUILD_FRAMEWORK
- `GDAL`_
- `HPX`_
- `Boost`_
- `HDF5`_
- `fmt`_
- `nlohmann-json3-dev`_

LUE_BUILD_VIEW
- `GDAL`_
- `HPX`_
- `Boost`_
- `HDF5`_
- `fmt`_
- `nlohmann-json3-dev`_
- `docopt`_
- `GLFW`_
- `ImGui`_

LUE_DATA_MODEL_WITH_PYTHON_API and LUE_FRAMEWORK_WITH_PYTHON_API
- `GDAL`_
- `HPX`_
- `Boost`_
- `HDF5`_
- `fmt`_
- `nlohmann-json3-dev`_
- `NumPy`_
- `Pybind11`_

LUE_DATA_MODEL_WITH_UTILITIES
- `GDAL`_
- `HPX`_
- `Boost`_
- `HDF5`_
- `fmt`_
- `nlohmann-json3-dev`_
- `docopt`_

Comment (kordejong) [reply document/doc/code/index.rst:1]: I only meant that you could add this to the existing version, not to list the dependencies per target. This amount of detail is not needed. Just revert this file to the original version, but add a comment that the dependencies depend on the targets being built.

**Review by kordejong — APPROVED** [review]

**Pull request merged**